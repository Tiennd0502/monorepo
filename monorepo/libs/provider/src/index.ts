export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
