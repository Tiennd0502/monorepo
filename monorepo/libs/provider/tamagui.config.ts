import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
