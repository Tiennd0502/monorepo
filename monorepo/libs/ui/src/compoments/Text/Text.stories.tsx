import { Meta, StoryObj } from '@storybook/react';

import Text from '.';

const meta: Meta<typeof Text> = {
  title: 'Components/Text',
  component: Text,
};

export default meta;

type Story = StoryObj<typeof Text>;

export const Default: Story = {
  args: {
    children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
};

export const Error: Story = {
  args: {
    error: true,
    children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
};

export const Bold: Story = {
  args: {
    bold: true,
    children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
};
