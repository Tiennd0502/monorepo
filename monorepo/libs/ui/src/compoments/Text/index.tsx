import { Text as TextBase, styled } from 'tamagui';

const Text = styled(TextBase, {
  color: '$textDefault',
  fontWeight: 'normal',
  fontFamily: '$body',
  allowFontScaling: false,

  defaultVariants: {
    size: 'medium',
  },

  variants: {
    error: {
      true: {
        color: '$errorPrimary',
      },
    },
    bold: {
      true: {
        fontWeight: 'bold',
        fontFamily: '$title',
      },
    },
    boldReversal: {
      true: {
        fontFamily: '$heading',
        color: '$textReversal',
        fontWeight: 'bold',
      },
    },

    size: {
      small: {
        fontSize: '$2.5',
        lineHeight: 12,
      },

      medium: {
        fontSize: '$3.5',
        lineHeight: 20,
      },

      extraMedium: {
        fontSize: '$4',
        lineHeight: 20,
      },

      large: {
        fontSize: '$4.5',
        lineHeight: 20,
      },

      extraLarge: {
        fontSize: '$6',
        lineHeight: 30,
      },

      huge: {
        fontSize: '$7.5',
        lineHeight: 41,
        fontWeight: 'bold',
      },
    },
  } as const,
});

export default Text;
