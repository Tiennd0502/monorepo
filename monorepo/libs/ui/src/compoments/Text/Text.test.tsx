import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import Text from '.';

const baseProps = {
  children: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.'
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Text {...props} />
    </TamaguiProvider>
  );
};

describe('Text', () => {
  it('should render Text component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Text component with prop error true', () => {
    const container = setup({error: true});

    expect(container).toMatchSnapshot();
  });

  it('should render Text component with size large', () => {
    const container = setup({size: 'large'});

    expect(container).toMatchSnapshot();
  });
});
