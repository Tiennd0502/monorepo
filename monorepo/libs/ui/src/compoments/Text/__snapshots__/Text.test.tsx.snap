// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render Text component 1`] = `
<Text
  allowFontScaling={false}
  style={
    {
      "color": "#999",
      "fontFamily": "NunitoSans_7pt-Medium",
      "fontSize": 14,
      "lineHeight": 20,
    }
  }
  suppressHighlighting={true}
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit.
</Text>
`;

exports[`Text should render Text component with prop error true 1`] = `
<Text
  allowFontScaling={false}
  style={
    {
      "color": "#eb5757",
      "fontFamily": "NunitoSans_7pt-Medium",
      "fontSize": 14,
      "lineHeight": 20,
    }
  }
  suppressHighlighting={true}
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit.
</Text>
`;

exports[`Text should render Text component with size large 1`] = `
<Text
  allowFontScaling={false}
  style={
    {
      "color": "#999",
      "fontFamily": "NunitoSans_7pt-Medium",
      "fontSize": 18,
      "lineHeight": 20,
    }
  }
  suppressHighlighting={true}
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit.
</Text>
`;
